bluemira/geometry/varied_offset.py

```diff
@@ -0,0 +1,165 @@
+# bluemira is an integrated inter-disciplinary design tool for future fusion
+# reactors. It incorporates several modules, some of which rely on other
+# codes, to carry out a range of typical conceptual fusion reactor design
+# activities.
+#
+# Copyright (C) 2022 M. Coleman, J. Cook, F. Franza, I.A. Maione, S. McIntosh, J. Morris,
+#                    D. Short
+#
+# bluemira is free software; you can redistribute it and/or
+# modify it under the terms of the GNU Lesser General Public
+# License as published by the Free Software Foundation; either
+# version 2.1 of the License, or (at your option) any later version.
+#
+# bluemira is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+# Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public
+# License along with bluemira; if not, see <https://www.gnu.org/licenses/>.
+"""
+Module containing functions to generate variable offset curves
+"""
+
+import numpy as np
+
+from bluemira.geometry.tools import find_clockwise_angle_2d, make_bspline
+from bluemira.geometry.wire import BluemiraWire
+
+
+def varied_offset(
```

Comment:
I think this should live in `tools.py` next to `offset_wire`. Alternatively both `offset` functions could live in `geometry/offset.py`; I have a feeling these won't be the only offset tools we end up with.

Thoughts on where this should live, @ivanmaione ?

Reply:
@hsaunders1904 sorry to keep moving the goalposts; I think given that this function is pretty specific, it would probably best live in `builders/EUDEMO/tools.py` for now - until such a time that (also, if) we can find a completely generic variable offset.

Reply:
IMHO this function is not so general to be placed in geometry. Its arguments are very specific to the build of a BB cross-section in a tokamak (or similar) and that strongly limits the generality of the intended function.
For the time being, I would propose to place this function "somewhere" in builders (since it just helps in building a particular component) and, in the future, implement something a little bit more general (at least for the input parameters) in geometry.